File: ios/RailwayWidget/Domain/Entity/WidgetRoute.swift
import Foundation

enum RailwaySystem: String, Codable {
    case tr = "TR"
    case hsr = "HSR"

    var prefix: String {
        switch self {
        case .tr: return "tr"
        case .hsr: return "hsr"
        }
    }
}

struct WidgetRoute: Codable, Equatable {
    let system: RailwaySystem
    let fromId: String
    let fromName: String
    let toId: String
    let toName: String
}

File: lib/features/widget_config/domain/entity/widget_route.dart
import 'dart:convert';

enum WidgetRailwaySystem { tr, hsr }

class WidgetRoute {
  final WidgetRailwaySystem system;
  final String fromId;
  final String fromName;
  final String toId;
  final String toName;

  const WidgetRoute({
    required this.system,
    required this.fromId,
    required this.fromName,
    required this.toId,
    required this.toName,
  });

  String toJson() => jsonEncode({
        'system': system == WidgetRailwaySystem.tr ? 'TR' : 'HSR',
        'fromId': fromId,
        'fromName': fromName,
        'toId': toId,
        'toName': toName,
      });
}
